test/jdk/java/awt/print/PrinterJob/LandscapeStackOverflow.java

```diff
@@ -35,14 +35,22 @@
 import java.awt.print.Printable;
 import java.awt.print.PrinterException;
 import java.awt.print.PrinterJob;
+import javax.print.PrintService;
 import javax.print.attribute.HashPrintRequestAttributeSet;
 import javax.print.attribute.PrintRequestAttributeSet;
 import javax.print.attribute.standard.OrientationRequested;
 
 public class LandscapeStackOverflow {
 
-    public static final void main( String[] parameters ) {
+    public static final void main( String[] parameters ) throws Exception {
+        PrintService[] printService = PrinterJob.lookupPrintServices();
```

Comment:
As far as I understand and comment in my previouse comment, if the lookupPrintServices() return empty array then the "PrinterJob.print()" will throw the same PrinterException("No print service found");

And if default printer is not set, and we have some other printers then "PrinterJob.print()" will select some non-default printer already, isn't it?

I meant that we just duplicate an existed checks, and both of them wll not work. If the system has some non-real printers like XPS, or "print to PDF" then such printers will be count  by the "PrinterJob.lookupPrintServices()" so no exception will be thrown and the test just hands suggestion the user to save the printed document somewhere. The hang occur exactly where it is reported in JBS "WPrinterJob._startDoc".

I can reproduce it on the freshly installed Windows.

Reply:
I'm not sure we can avoid this if user selects "print to pdf" as default printservice or if it is the only printservice available. Maybe converting it to manual is the way to go.

Reply:
We can update the documentation "testing.html" about the "printer" keyword and describe how to skip/run the tests marked by this keyword. or we can just close the bug as not-a-defect.